### PR TITLE
#144 Create methods to find euro protocols for view

### PR DESCRIPTION
**Pull request by aliquispassim**

### CODE REVIEW

**Review by SashaKorniichuk — APPROVED** [review]

File: CrudMicroservice.Domain/ApiModel/ResponseApiModels/EuroProtocolFullResponseModel.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+﻿using CrudMicroservice.Data.Entities;
+using System.Collections.Generic;
+
+namespace CrudMicroservice.Domain.ApiModel.ResponseApiModels
+{
+    public class EuroProtocolFullResponseModel
+    {
+        public EuroProtocolSimpleResponseModel EuroProtocolSimple { get; set; }
```
Comment (ICHTLAY) [inline CrudMicroservice.Domain/ApiModel/ResponseApiModels/EuroProtocolFullResponseModel.cs]: it should contain `EuroProtocolResponseModel`, not `EuroProtocolSimpleResponseModel`, and property should be named just as `EuroProtocol`

File: CrudMicroservice.Domain/ApiModel/ResponseApiModels/EuroProtocolFullResponseModel.cs [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+        public UserResponseModel UserDataSideA { get; set; }
+        public UserResponseModel UserDataSideB { get; set; }
+        public TransportResponseApiModel TransportSideA { get; set; }
+        public TransportResponseApiModel TransportSideB { get; set; }
+        public IEnumerable<Witness> Witnesses { get; set; }
```
Comment (ICHTLAY) [inline CrudMicroservice.Domain/ApiModel/ResponseApiModels/EuroProtocolFullResponseModel.cs]: it should contain `IEnumerable<WitnessResponseModel>`, not `IEnumerable<Witness>`

File: CrudMicroservice.Domain/Services/EuroProtocolService.cs [hunk 3]
```diff
@@ -101,5 +109,57 @@ public async Task<IEnumerable<CircumstanceResponseModel>> GetAllCircumstances()
...
+        public async Task<EuroProtocolFullResponseModel> GetEuroProtocolBySerialNumber(string serialNumber)
+        {
+            var euroProtocol = await _euroProtocols.GetByFilterAsync(x => x.SerialNumber == serialNumber);
+
+            if (euroProtocol == null)
+            {
+                throw new RestException(HttpStatusCode.NotFound, Resources.ResourceManager.GetString("EuroProtocolNotFound"));
+            }
+
+            var mappedEuroProtocolSimpleResponseModel = _mapper.Map<EuroProtocolSimpleResponseModel>(euroProtocol);
```
Comment (ICHTLAY) [inline CrudMicroservice.Domain/Services/EuroProtocolService.cs]: you should map to `EuroProtocolResponseModel`, not to `EuroProtocolSimpleResponseModel`

**Review by ICHTLAY — CHANGES_REQUESTED** [review]

**Review by pinkevmladchy — APPROVED** [review]
but resolve conversation which detected by Ivan

File: CrudMicroservice.Domain/ApiModel/ResponseApiModels/EuroProtocolFullResponseModel.cs (hunk 2 above)
Comment (aliquispassim) [reply CrudMicroservice.Domain/ApiModel/ResponseApiModels/EuroProtocolFullResponseModel.cs]: okay, but `EuroProtocolResponseModel` already contains `IEnumerable<WitnessResponseModel> Witnesses`. maybe I should remove this field from `EuroProtocolFullResponseModel`?

**Review by ICHTLAY — APPROVED** [review]

**Pull request merged**